calico_node/filesystem/startup.py

```diff
@@ -261,7 +261,7 @@ def main():
     # This is required because the confd templates expect to be able to fill in
     #  the IP by fetching it from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("IP=" + str(ip))
```

Comment:
It is nicer to use IPAddress consistently when possible - but might be sensible to be pragmatic here since the other methods are not part of a public API.